pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -282,6 +282,11 @@ func (c *Cluster) startMons(targetCount int) error {
 	}
 
 	logger.Debugf("mon endpoints used are: %s", FlattenMonEndpoints(c.ClusterInfo.Monitors))
+
+	// Check if there are orphaned mon resources that should be cleaned up at the end of a reconcile.
+	// There may be orphaned resources if a mon failover was aborted.
+	c.removeOrphanMonResources()
```

Comment:
Could we try even earlier? As soon as we enter the mon code? If the reconcile mon code fails, we will never reach this right?

Reply:
Cleaning up these orphaned resources is really only a nice-to-have, so running it with each reconcile is fine. It's not critical, so it doesn't need to run every time though. And actually in some disaster scenarios (if the mon deployments are missing), it could make things worse by calling this first before the mon deployments have been re-created.